WooCommerce/Classes/ViewRelated/Editor/EditorFactory.swift

```diff
@@ -0,0 +1,22 @@
+import Yosemite
+
+protocol Editor {
+    typealias OnContentSave = (_ content: String) -> Void
+    var onContentSave: OnContentSave? { get }
+
+    init(content: String?)
+}
+
+/// This class takes care of instantiating the correct editor based on the App settings, feature flags,
+/// etc.
+///
+class EditorFactory {
```

Comment:
Have you considered exposing this factory through the ServiceLocator? It seems like something that much the worth setting up for mocking.

Reply:
Also you know I have a thing with marking classes as final 😛

Reply:
I took a look at `ServiceLocator` and it felt a bit out of place for a specific UI factory. Do you think it's worth creating a protocol for `EditorFactory` and add it to `ServiceLocator`? Were you thinking about mocking the `EditorFactory` to return different editor UI to test the interactions? I thought the `EditorFactory` is pretty local with probably few entry points, so if we want to mock it we can DI the factory manually in those places

Reply:
It seems to me like the choice of editor is suitable for being a global setting (and in the past that's how it was on WordPress), that's why I suggested making it global. I am not sure that makes sense thought...